### PR TITLE
Add equations for gray radiation

### LINKED ISSUE #329: Add  equations for gray radiation

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Closes #329 

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by szy21** [comment]
@tapios Could you take a look when you have time? Thanks.

File: docs/src/Optics.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# RRTMGP Optics
+
+`compute_optical_props!` computes the optical properties and source functions of an atmospheric state. For the two-stream approximation, the optical properties include the optical depth (``\tau``), single-scattering albedo (``\omega_0``), and asymmetry parameter (``g``).
+
+## Gray atmosphere optics
+
+The gray atmosphere is an approximation of the radiative properties of the atmosphere based on the assumption that the optical property is independent of wavelength. The default optical property for gray atmosphere is different for longwave and shortwave radiation.
```
Comment (tapios) [inline docs/src/Optics.md:7]:
```suggestion
The gray atmosphere (more precisely, semi-gray atmosphere) is an approximation of the radiative properties of the atmosphere that assumes the optical properties are independent of wavelength, separately in the longwave and shortwave bands. 
```

File: docs/src/Optics.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+### Longwave
+
+The optical depth for longwave radiation follows [schneider2004](@cite):
+
+```math
+\begin{align}
+\tau = (\alpha d_0 (\frac{p}{p_0})^\alpha / p) \Delta p
```
Comment (tapios) [inline docs/src/Optics.md:15]:
```suggestion
\tau = \alpha d_0 \left(\frac{p}{p_0}\right)^\alpha \frac{\Delta p}{p} .
```
(I don't fully understand this expression. Why is there \Delta p/p here, and alpha out front? It shouldn't be there if it's optical depth, I think. Maybe you want this to be a layerwise transmissivity?)

File: docs/src/Optics.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+\end{align}
+```
+
+where ``\alpha = 3.5`` controls the lapse rate of the troposphere. ``p`` is the pressure of the atmosphere layer and ``p_0 = 1000~\mathrm{hPa}`` is the reference pressure. ``\Delta p`` is the pressure thickness of the atmosphere layer. ``d_0`` is calculated from the radiative equilibrium temperature:
```
Comment (tapios) [inline docs/src/Optics.md:19]:
```suggestion
Here, ``\alpha``  is ratio of the pressure scale height to the partial-pressure scale height of the infrared absorber (``\alpha=3.5`` for water vapor); ``p`` is the pressure of the atmosphere layer, and ``p_0 = 1000~\mathrm{hPa}`` is the reference pressure; ``\Delta p`` is the pressure thickness of an atmosphere layer; and the optical depth at ``p_0``, ``d_0``, is calculated from the radiative equilibrium temperature:
```

File: docs/src/Optics.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
```
Comment (tapios) [inline docs/src/Optics.md:20]: Make clear d_0 depends on latitude.

File: docs/src/Optics.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+```math
+\begin{align}
+T_s = T_e + \Delta T (\frac{1}{3} - sin^2\phi) //
```
Comment (tapios) [inline docs/src/Optics.md:23]:
```suggestion
T_s = T_e + \Delta T \left(\frac{1}{3} - \sin^2\phi\right) \\
```
This is hard to read. Break the line after T_s.

File: docs/src/Optics.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+d_0 = (\frac{T_s}{T_t})^4 - 1
```
Comment (tapios) [inline docs/src/Optics.md:24]:
```suggestion
d_0 = \left(\frac{T_s}{T_t}\right)^4 - 1.
```

File: docs/src/Optics.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+\end{align}
+```
+
+where ``T_e`` is the global mean surface temperature, ``T_t`` is the temperature at the top of the atmosphere, and ``\Delta T`` is the equator-to-pole temperature difference in radiative equilibrium. The default values are ``T_e = 300~\mathrm{K}``, ``T_t = 200~\mathrm{K}``, and ``\Delta T = 60~\mathrm{K}``. ``\phi`` is the latitude.
```
Comment (tapios) [inline docs/src/Optics.md:28]:
```suggestion
Here, ``T_e`` is the global mean surface temperature in radiative equilibrium, ``T_t`` is the temperature at the top of the atmosphere, and ``\Delta T`` is the equator-to-pole temperature difference in radiative equilibrium. The default values are ``T_e = 300~\mathrm{K}``, ``T_t = 200~\mathrm{K}``, and ``\Delta T = 60~\mathrm{K}``. ``\phi`` is the latitude.
```
Move \phi for latitude up to make clear d_0 depends on it.

File: docs/src/Optics.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+The source function for longwave radiation is calculated as ``S = \sigma T^4 / \pi``, where ``T`` is the air temperature and ``\sigma`` is the Stefan–Boltzmann constant.
+
+### Shortwave
+
+The optical depth for shortwave radiation is calculated as:
```
Comment (tapios) [inline docs/src/Optics.md:34]: Same as above. This doesn't look like an optical depth.

**Review by tapios — COMMENTED** [review]
Please have a look at the 2004 paper (and the one with Paul in 2008 for shortwave) and correct the documentation where needed (for optical depth).

**Comment by szy21** [comment]
> Please have a look at the 2004 paper (and the one with Paul in 2008 for shortwave) and correct the documentation where needed (for optical depth).

Right, the optical depth in the code is actually `d \tau / dp * \Delta p`. I don't know why, but maybe this gives more accurate results when you integrate in RTE? Would you like me to have the actual analytical optical depth in the docs (but it will be different from the code)?

**Comment by tapios** [comment]
It's not optical depth (probably transmissivity).

**Comment by szy21** [comment]
We call it optical depth [here](https://github.com/CliMA/RRTMGP.jl/blob/851f13ee1396745a8856cb75aa418657d9b99e8d/src/optics/GrayOpticsKernels.jl#L99). Should we change it then?

**Comment by tapios** [comment]
Probably good to make accurate. Pls check what RRTMG does.

**Comment by szy21** [comment]
RRTMGP calls it optical depth too (Note they don't have gray radiation). From the RTE it seems to be the optical depth of the layer. I don't know if there is a better name for it?

**Comment by szy21** [comment]
Actually I think optical thickness makes sense.

**Comment by tapios** [comment]
Maybe call it layerwise optical thickness or something like it, to make sure it's a differential optical thickness in a layer.

**Comment by szy21** [comment]
Ok, I'll update this documentation first, then update the comment in the code

**Comment by szy21** [comment]
@tapios Would you mind reviewing it again? Thanks.

File: docs/src/Optics.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
+# RRTMGP Optics
+
+`compute_optical_props!` computes the optical properties and source functions given an atmospheric state. For the two-stream approximation, the optical properties include the optical thickness of the atmosphere layer (``\tau``), single-scattering albedo (``\omega_0``), and asymmetry parameter (``g``).
+
+## Gray atmosphere optics
+
+The gray atmosphere (more precisely, semi-gray atmosphere) is an approximation of the radiative properties of the atmosphere that assumes the optical properties are independent of wavelength, separately in the longwave and shortwave bands.
```
Comment (tapios) [inline docs/src/Optics.md:7]:
```suggestion
The gray atmosphere (more precisely, semi-gray atmosphere) approximates the radiative properties of the atmosphere by taking the optical properties to be independent of wavelength, separately in the longwave and shortwave bands. 
```

File: docs/src/Optics.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+The optical depth (``d``) for longwave radiation follows [schneider2004](@cite):
+
+```math
+\begin{align}
+d(\phi, p) = d_0(\phi) (\frac{p}{p_0})^\alpha
+\end{align}
+```
+
+Here, ``\alpha``  is ratio of the pressure scale height to the partial-pressure scale height of the infrared absorber (``\alpha=3.5`` for water vapor); ``\phi`` is the latitude; ``p`` is the pressure of the atmosphere layer, and ``p_0 = 1000~\mathrm{hPa}`` is the reference pressure; and the optical depth at ``\phi`` and ``p_0``, ``d_0``, is calculated from the latitude-dependent radiative equilibrium temperature:
```
Comment (tapios) [inline docs/src/Optics.md:19]:
```suggestion
Here, ``\alpha``  is ratio of the pressure scale height to the partial-pressure scale height of the infrared absorber (``\alpha=3.5`` for water vapor); ``\phi`` is the latitude; ``p`` is the pressure, and ``p_0 = 1000~\mathrm{hPa}`` is the reference pressure. The optical depth at ``\phi`` and ``p_0``, ``d_0``, is calculated from the latitude-dependent radiative equilibrium temperature ``T_s(\phi)``:
```

File: docs/src/Optics.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+```math
+\begin{align}
+T_s(\phi) = T_e + \Delta T \left(\frac{1}{3} - \sin^2\phi\right) \\
+d_0(\phi) = \left(\frac{T_s(\phi)}{T_t}\right)^4 - 1.
+\end{align}
+```
+
+Here, ``T_e`` is the global mean surface temperature in radiative equilibrium, ``T_t`` is the temperature at the top of the atmosphere, and ``\Delta T`` is the equator-to-pole temperature difference in radiative equilibrium. The default values are ``T_e = 300~\mathrm{K}``, ``T_t = 200~\mathrm{K}``, and ``\Delta T = 60~\mathrm{K}``.
```
Comment (tapios) [inline docs/src/Optics.md:28]:
```suggestion
Here, ``T_e`` is the global-mean surface temperature in radiative equilibrium, ``T_t`` is the temperature at the top of the atmosphere, and ``\Delta T`` is the equator-to-pole temperature difference in radiative equilibrium. The default values are ``T_e = 300~\mathrm{K}``, ``T_t = 200~\mathrm{K}``, and ``\Delta T = 60~\mathrm{K}``.
```

File: docs/src/Optics.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+The optical thickness of the atmosphere layer (the differential optical depth) is:
```
Comment (tapios) [inline docs/src/Optics.md:30]:
```suggestion
The optical thickness of an atmosphere layer (the differential optical depth) of pressure thickness ``\Delta p`` is
```

File: docs/src/Optics.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+```math
+\begin{align}
+\tau = \alpha d_0 \left(\frac{p}{p_0}\right)^\alpha \frac{\Delta p}{p}
```
Comment (tapios) [inline docs/src/Optics.md:33]:
```suggestion
\tau = \alpha d_0 \left(\frac{p}{p_0}\right)^\alpha \frac{\Delta p}{p}.
```

File: docs/src/Optics.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+\end{align}
+```
+where ``\Delta p`` is the pressure thickness of the atmosphere layer.
```
Comment (tapios) [inline docs/src/Optics.md:36]:
```suggestion
```

File: docs/src/Optics.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+The source function for longwave radiation is calculated as ``S = \sigma T^4 / \pi``, where ``T`` is the air temperature and ``\sigma`` is the Stefan–Boltzmann constant.
+
+### Shortwave
+
+The optical depth (``d``) for shortwave radiation follows [ogorman2008](@cite):
```
Comment (tapios) [inline docs/src/Optics.md:42]: I think this was also in Frierson & Held (2006 or 2007). Can you please check and add the reference if needed?

File: docs/src/Optics.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+
+```math
+\begin{align}
+d(p) = \tau_0 \left(\frac{p}{p_0}\right)^2
+\end{align}
+```
+
+where ``p`` is the pressure of the atmosphere layer and ``p_0 = 1000~\mathrm{hPa}`` is the reference pressure. The default value for ``\tau_0`` is 0.22. 
+
+The optical thickness of the atmosphere layer is:
```
Comment (tapios) [inline docs/src/Optics.md:52]:
```suggestion
The differential optical thickness of an atmosphere layer of pressure thickness ``\Delta p`` is
```

File: docs/src/Optics.md [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+
+```math
+\begin{align}
+\tau(p) = 2 \tau_0 \frac{p}{p_0} \frac{\Delta p}{p_0}
+\end{align}
+```
+
+where ``\Delta p`` is the pressure thickness of the atmosphere layer.
```
Comment (tapios) [inline docs/src/Optics.md:60]:
```suggestion
```

File: docs/src/Optics.md [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+\end{align}
+```
+
+where ``p`` is the pressure of the atmosphere layer and ``p_0 = 1000~\mathrm{hPa}`` is the reference pressure. The default value for ``\tau_0`` is 0.22. 
+
+The optical thickness of the atmosphere layer is:
+
+```math
+\begin{align}
+\tau(p) = 2 \tau_0 \frac{p}{p_0} \frac{\Delta p}{p_0}
```
Comment (tapios) [inline docs/src/Optics.md:56]:
```suggestion
\tau(p) = 2 \tau_0 \frac{p}{p_0} \frac{\Delta p}{p_0}.
```

**Review by tapios — APPROVED** [review]
Looks good.

File: docs/src/Optics.md (hunk 15 above)
Comment (szy21) [reply docs/src/Optics.md:41]: Frierson (2007) has a shortwave optical depth, but the exact formulation is different from what we have here. In his paper `\tau = \tau_0 (p/p_0)^4`.

**Pull request merged**